pkg/controllers/rollout/override_test.go

```diff
@@ -0,0 +1,905 @@
+			}
+			for i := range tc.roList {
+				objects = append(objects, &tc.roList[i])
+			}
+			fakeClient := fake.NewClientBuilder().
+				WithScheme(scheme).
+				WithObjects(objects...).
+				Build()
+			r := Reconciler{
+				Client:          fakeClient,
+				InformerManager: &fakeInformer,
+			}
+			gotCRO, gotRO, err := r.getAllOverrides(context.Background(), crpName, tc.master)
+			if err != nil {
+				t.Fatalf("getAllOverrides() failed, got err %v, want no err", err)
+			}
+			options := []cmp.Option{
+				cmpopts.IgnoreFields(metav1.ObjectMeta{}, "ResourceVersion"),
+				cmpopts.SortSlices(func(o1, o2 fleetv1alpha1.ClusterResourceOverride) bool {
+					return o1.Name < o2.Name
+				}),
+				cmpopts.SortSlices(func(o1, o2 fleetv1alpha1.ResourceOverride) bool {
+					if o1.Namespace == o2.Namespace {
+						return o1.Name < o2.Name
+					}
+					return o1.Namespace < o2.Namespace
+				}),
+				cmpopts.EquateEmpty(),
+			}
+			if diff := cmp.Diff(tc.wantCRO, gotCRO, options...); diff != "" {
+				t.Errorf("getAllOverrides() returned clusterResourceOverrides mismatch (-want, +got):\n%s", diff)
+			}
+			if diff := cmp.Diff(tc.wantRO, gotRO, options...); diff != "" {
+				t.Errorf("getAllOverrides() returned resourceOverrides mismatch (-want, +got):\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestGetLatestOverridesPerBinding(t *testing.T) {
+	tests := []struct {
+		name    string
+		cluster *clusterv1beta1.MemberCluster
+		croList []*fleetv1alpha1.ClusterResourceOverride
+		roList  []*fleetv1alpha1.ResourceOverride
+		wantCRO []string
+		wantRO  []fleetv1beta1.NamespacedName
+		wantErr error
+	}{
+		{
+			name: "cluster not found",
+			cluster: &clusterv1beta1.MemberCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster-not-exist",
+				},
+			},
+			wantErr: controller.ErrExpectedBehavior,
+		},
+		{
+			name: "empty overrides",
+			cluster: &clusterv1beta1.MemberCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster-1",
+				},
+			},
+			wantCRO: []string{},
+			wantRO:  []fleetv1beta1.NamespacedName{},
+		},
+		{
+			name: "matched overrides with empty cluster label",
+			cluster: &clusterv1beta1.MemberCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster-1",
+				},
+			},
+			croList: []*fleetv1alpha1.ClusterResourceOverride{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "cro-1",
+					},
+					Spec: fleetv1alpha1.ClusterResourceOverrideSpec{
+						ClusterResourceSelectors: []fleetv1beta1.ClusterResourceSelector{
+							{
+								Group:   "",
+								Version: "v1",
+								Kind:    "Namespace",
+								Name:    "name-1",
+							},
+						},
+						Policy: &fleetv1alpha1.OverridePolicy{
+							OverrideRules: []fleetv1alpha1.OverrideRule{
+								{
+									JSONPatchOverrides: []fleetv1alpha1.JSONPatchOverride{
+										{
+											Operator: fleetv1alpha1.JSONPatchOverrideOpAdd,
+											Path:     ".meta",
+											Value:    "abc",
+										},
+									},
+								},
```

Comment:
why does this field matter in this test?

Reply:
it does not. removed